archivessnake/get_labels.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
+
+import argparse, configparser, json, csv
+import os.path
+
+from asnake.aspace import ASpace
+
+class LabelPrinter:
+    def __init__(self, resource):
+        config = configparser.ConfigParser()
+        config.read('local_settings.cfg')
+        self.aspace = ASpace(baseurl=config.get('ArchivesSpace', 'baseURL'), username=config.get('ArchivesSpace', 'user'), password=config.get('ArchivesSpace', 'password'))
+        self.resource = self.aspace.repositories(2).resources(resource)
+
+    def run(self):
+        label_data = []
+        resource_title = self.resource.title
+        resource_id = self.get_id()
+        for obj in self.resource.tree.walk:
+            if len(obj.instances):
+                parent = self.get_parent(obj)
+                container = self.get_containers(obj)
+                print(resource_title, resource_id, parent, container)
+                label_data.append([resource_title, resource_id, parent, container])
+        print("Raw data", len(label_data))
+        print("Deduplicated data", len(set(tuple(row) for row in label_data)))
+        label_data = set(tuple(row) for row in label_data)
+        self.make_csv(label_data)
+
+    def make_csv(self, label_data):
+        with open('box_labels.csv', 'w') as csvfile:
+            writer = csv.writer(csvfile)
+            writer.writerow(['ResourceTitle', 'ResourceID', 'ParentTitle', 'Container'])
+            writer.writerows(label_data)
+
+    def get_id(self):
+        """
+        Gets and returns the id of a resource record.
+        """
+        id = self.resource.id_0
+        return id
```

Comment:
I would think something like:

```
number = []
for n in range(0,2):
  try:
    number.append(getattr(self.resource, "id_{}".format(2))
  except KeyError:
    break
":".join(number)
```
Not sure if that's catching the right exception, but something along those lines...